src/network.c

```diff
@@ -97,3 +101,20 @@ int create_epoll_socket() {
 
     return epoll_descriptor;
 }
+
+int full_message_availiable(int socket) {
+    int message[MAX_SIZE_MESSAGE_INT];
+    int message_len_recv = 0; // Message length received.
+    int message_len = 0; // The intended message length.
+
+    // https://pubs.opengroup.org/onlinepubs/007904975/functions/recv.html
+    // Peek the message at the socket.
+    message_len_recv = recv(socket, message, MAX_SIZE_MESSAGE_INT*4, MSG_PEEK);
+
+    if (message_len < 4) {
```

Comment:
```suggestion
    if (message_len_recv >=4) {
```
I think that's what you meant? message_len is always false here

Reply:
Yep, facepalm here.